From 7beffea86bb20ac9c303f8507fe21def1d035e3b Mon Sep 17 00:00:00 2001
From: "Luke A. Guest" <laguest@archeia.com>
Date: Sun, 29 Jan 2017 11:20:32 +0000
Subject: [PATCH 04/14] Fix kernel module installation location using dkms

---
 dkms.conf | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/dkms.conf b/dkms.conf
index ad68a3a..d8414a7 100644
--- a/dkms.conf
+++ b/dkms.conf
@@ -9,13 +9,13 @@ PRE_BUILD="pre-build.sh $kernelver"
 
 BUILT_MODULE_NAME[0]="amdgpu"
 BUILT_MODULE_LOCATION[0]="amd/amdgpu"
-DEST_MODULE_LOCATION[0]="/updates"
+DEST_MODULE_LOCATION[0]="/kernel/drivers/gpu/drm/amd/amdgpu"
 
 BUILT_MODULE_NAME[1]="amdttm"
 BUILT_MODULE_LOCATION[1]="ttm"
-DEST_MODULE_LOCATION[1]="/updates"
+DEST_MODULE_LOCATION[1]="/kernel/drivers/gpu/drm/amd/amdgpu"
 
 BUILT_MODULE_NAME[2]="amdkcl"
 BUILT_MODULE_LOCATION[2]="amd/amdkcl"
-DEST_MODULE_LOCATION[2]="/updates"
+DEST_MODULE_LOCATION[2]="/kernel/drivers/gpu/drm/amd/amdgpu"
 POST_REMOVE="post-remove.sh $kernelver"
-- 
2.7.3

