From f4cb730d8d3944a050bfabfff9a520ab5d6abb1b Mon Sep 17 00:00:00 2001
From: "Luke A. Guest" <laguest@archeia.com>
Date: Fri, 3 Feb 2017 15:38:38 +0000
Subject: [PATCH 07/14] Add cast to size_t in min() call, for kernel v4.10.

---
 amd/amdgpu/amdgpu_kms.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/amd/amdgpu/amdgpu_kms.c b/amd/amdgpu/amdgpu_kms.c
index b7b51ae..1228b01 100644
--- a/amd/amdgpu/amdgpu_kms.c
+++ b/amd/amdgpu/amdgpu_kms.c
@@ -591,8 +591,13 @@ static int amdgpu_info_ioctl(struct drm_device *dev, void *data, struct drm_file
 
 			bios = adev->bios + bios_offset;
 			return copy_to_user(out, bios,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 10, 0)
+				min((size_t)size, (size_t)(bios_size - bios_offset)))
+					? -EFAULT : 0;
+#else
 				min((size_t)size, bios_size - bios_offset))
 					? -EFAULT : 0;
+#endif /* LINUX_VERSION_CODE >= KERNEL_VERSION(4, 10, 0) */
 		}
 		default:
 			DRM_DEBUG_KMS("Invalid request %d\n",
-- 
2.7.3

